src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int CAPACITY = 16;
+    private static final int RATE = 4;
+    private Object[] array = new Object[CAPACITY];
+    private int size = 0;
+
+    public ArrayList() {
+
+        T elementData = (T) new Object[CAPACITY];
+    }
+
     @Override
     public void add(T value) {
+        if (size == array.length - 1) {
+            resize(array.length * 2);
```

Comment:
size of a new array should be X1,5 as previous

Reply:
done